### PR TITLE
setup.py: fix error handeling when GMP is missing

### PR DESCRIPTION
**Pull request by lancelotsix**
For tests to run without gmp, original configuration files (i.e.
`pythran/pythran-*.cfg`) have to be modified, which is quite unsafe.

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@lancelotsix : can you check if #515 fixes your issue? It has been reviewed and is almost ready to be merged!

**Comment by lancelotsix** [comment]
OK, I'll do that

File: setup.py [hunk 1]
```diff
@@ -125,23 +126,30 @@ def detect_gmp(self):
...
             print('Failed to compile GMP source, disabling long support.')
-            for cfg in glob.glob(os.path.join(pythrandir, "pythran-*.cfg")):
-                with open(cfg, "r+") as cfg:
-                    content = cfg.read()
-                    content = content.replace('USE_GMP', '')
-                    content = content.replace('gmp gmpxx', '')
-                    cfg.seek(0)
-                    cfg.write(content)
-        map(os.remove, objs + srcs + [exe])
+            for pythrandir in (os.path.join(self.build_lib, "pythran"),
```
Comment (serge-sans-paille) [inline setup.py]: @pbrunet : I prefer this modification rather than modifying 'pythran'

File: setup.py [hunk 2]
```diff
@@ -125,23 +126,30 @@ def detect_gmp(self):
...
+                               os.path.join(os.path.dirname(__file__),
+                                            "pythran")):
```
Comment (serge-sans-paille) [inline setup.py]: @lancelotsix why do you modify the second version of the cfg. The one that lies in build should be enough, right?

Comment (lancelotsix) [inline setup.py]: The configs in `self.build_lib` are staged for install while the one in `.` (the one updated in #515) are the ones that are loaded when you run `python setup.py test`. I think around here, you want to update both (even it is kind of a bad idea to update a version controlled file…).

**Pull request closed without merging**